src/battle_ai_main.c

```diff
@@ -4871,6 +4891,15 @@ static s16 AI_CheckViability(u8 battlerAtk, u8 battlerDef, u16 move, s16 score)
         //break;
     //case EFFECT_SKY_DROP
         //break;
+    case EFFECT_JUNGLE_HEALING:
+        if (ShouldRecover(battlerAtk, battlerDef, move, 25)
+         || ShouldRecover(battlerAtk, BATTLE_PARTNER(battlerDef), move, 25)
+         || ShouldRecover(BATTLE_PARTNER(battlerAtk), battlerDef, move, 25)
+         || ShouldRecover(BATTLE_PARTNER(battlerAtk), BATTLE_PARTNER(battlerDef), move, 25)
```

Comment:
Considering this is the `ShouldRecover` function ported from an older version of the CFRU, this logic is completely wrong. The function determines if the attacker should heal against a foe. Considering the partner is very much not an enemy, this will make the AI make the wrong decision. You want two ShouldRecovers. One with the attacker vs the current target, and one with the partner vs the current target.

Reply:
> You want two ShouldRecovers. One with the attacker vs the current target, and one with the partner vs the current target.

But then it's not completely wrong, is it? 🤔 
You want me to delete `|| ShouldRecover(battlerAtk, BATTLE_PARTNER(battlerDef), move, 25)` and `|| ShouldRecover(BATTLE_PARTNER(battlerAtk), BATTLE_PARTNER(battlerDef), move, 25)`, and leave the other 2, is that correct or am I misunderstanding? :eyes:

Reply:
Sorry, I don't know why I read it as something completely different haha
Yeah, that's much better. You don't need to check both targets because you have a target loop in doubles and the partners will be accounted for then.